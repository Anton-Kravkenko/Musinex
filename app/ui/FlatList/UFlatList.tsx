import { Ionicons } from '@expo/vector-icons/'
import React from 'react'
import { FlatList, FlatListProps, ListRenderItem, Pressable, View } from 'react-native'
import Title from '../title/title'

export interface IFlatList<T> extends Omit<FlatListProps<T>, "renderToHardwareTextureAndroid" | "bounces"> {
	data: T[]
	renderItem: ListRenderItem<T>
	header?: boolean
	headerText?: string
	wrapClassNames?: string
	headerNavigate?: () => void
}

const UFlatList = <T,>({ data, renderItem, header, headerNavigate, wrapClassNames, headerText, ...rest }: IFlatList<T>) => {
	return (
		<View className={wrapClassNames}>
			{header && headerText ? (
				<View className='flex-row justify-between items-center mb-3'>
					<Title text={headerText} size={25} fontFamily={'Montserrat_500Medium'} />
					{data.length > 5 ?
					<Pressable onPress={headerNavigate} className='flex-row items-center'>
						<Title text={'See More'} size={25} fontFamily={'Montserrat_500Medium'} classNames='mr-1 mb-1' />
						<Ionicons name='ios-arrow-forward' size={20} color='white' />
					</Pressable>
					: null}
				</View>
			) : null}
			<FlatList
				data={data}
				renderItem={renderItem}
				renderToHardwareTextureAndroid={true}
				bounces={false}
				{...rest}
			/>
		</View>
	)
}

export default UFlatList
