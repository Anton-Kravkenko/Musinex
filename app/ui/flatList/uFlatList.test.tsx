import UFlatList from '@/ui/flatList/uFlatList'
import { render } from '@testing-library/react-native'
import { Pressable, View } from 'react-native'

const data = [
	{
		id: 1,
		name: 'name1'
	},
	{
		id: 2,
		name: 'name2'
	},
	{
		id: 3,
		name: 'name3'
	}
]

describe('uFlatList', () => {
	it('test with no data', () => {
		const { getByTestId } = render(
			<UFlatList data={[]} renderItem={() => <View />} />
		)
		expect(getByTestId('uFlatList')).toBeNull()
	})
	it('test with data and data.length', () => {
		const { getByTestId } = render(
			<UFlatList data={data} renderItem={() => <Pressable />} />
		)
		expect(getByTestId('uFlatList')).toBeTruthy()
	})
	it('test style', () => {})
	it('headerText', () => {})
})
