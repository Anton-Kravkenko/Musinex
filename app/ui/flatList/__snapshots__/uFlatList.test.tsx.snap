// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uFlatList Render 1`] = `
<View
  style={
    {
      "marginTop": 0,
    }
  }
>
  <RCTScrollView
    alwaysBounceHorizontal={false}
    alwaysBounceVertical={false}
    bounces={false}
    contentContainerStyle={
      [
        {
          "paddingBottom": 130,
        },
        undefined,
      ]
    }
    data={
      [
        {
          "id": 1,
          "name": "name1",
        },
        {
          "id": 2,
          "name": "name2",
        },
        {
          "id": 3,
          "name": "name3",
        },
      ]
    }
    decelerationRate="fast"
    getItem={[Function]}
    getItemCount={[Function]}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={true}
    renderItem={[Function]}
    renderToHardwareTextureAndroid={true}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    style={
      [
        {
          "height": "auto",
          "paddingLeft": 0,
        },
        {
          "backgroundColor": "red",
        },
      ]
    }
    testID="uFlatList"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
